src/main/java/games/strategy/util/Version.java

```diff
@@ -198,11 +196,15 @@ public String toStringFull() {
    * Creates a complete version string with the given separator, even if some version numbers are 0.
    */
   public String toStringFull(final char separator) {
-    return Joiner.on(separator).join(m_major, m_minor, m_point, m_micro);
+    return Joiner.on(separator).join(m_major, m_minor, m_point, m_micro == Integer.MAX_VALUE ? "dev" : m_micro);
```

Comment:
I'm curious why the replacement of `max_value` to dev is useful, what's the purpose behind this? The 'magic' value of `MAX_VALUE` could be a bit tricky to remember exactly, wondering if we can lop off the behavior if possible.

Reply:
Maybe it helps when we think of the case when version numbers are being compared.
In this case the "dev" versions would be always the newer version compared to stable which is a good thing...

Reply:
When an installer is generated in the build environment, it appends ".dev" to the version string. An argument (spurious IMO) was made that this needed to be handled rather than tolerating a console pop up.

Reply:
> When an installer is generated in the build environment, it appends ".dev" to the version string

Small clarification:

* If the Gradle project property `engineVersion` is set, it is used as-is.  (The Travis build [sets this project property](https://github.com/triplea-game/triplea/blob/master/.travis.yml#L21-L22) to the value of the `engine_version` property from _game\_engine.properties_ concatenated with `.` concatenated with the Travis build number.)
* Otherwise, it uses the value of the `engine_version` property from _game\_engine.properties_ concatenated with `.dev`.